package/MDAnalysis/analysis/contacts.py

```diff
@@ -164,6 +176,199 @@
+
+
+class Contacts(AnalysisBase):
+    """Calculate fraction of native contacts (Q) from a trajectory
+
+    Inputs
+    ------
+    Two string selections for the contacting AtomGroups,
+    the groups could be protein-lipid or protein-protein.
+
+    Use two reference AtomGroups to obtain reference distances (r0)
+    for the cotacts.
+
+    Methods available
+    -----------------
+    Supports either hard-cutoff or soft-cutoff (Best-Hummer like [1]_)
+    contacts.
+
+
+    Parameter choices
+    -----------------
+    There are recommendations and not strict orders.
+    These parameters should be insensitive to small changes.
+    * For all-atom simulations, radius = 4.5 A and lambda_constant = 1.8 (unitless)
+    * For coarse-grained simulations, radius = 6.0 A and lambda_constant = 1.5 (unitless)
+
+    """
+    def __init__(self, u, selection, refgroup, method="cutoff", radius=4.5,
+                 start=None, stop=None, step=None, **kwargs):
+        """
+
+        Parameters
+        ----------
+        u: Universe
+            trajectory
+        selection: tuple(string, string)
+            two contacting groups that change over time
+        refgroup: tuple(AtomGroup, AtomGroup)
+            two contacting groups in their reference conformation
+        radius: float, optional (4.5 Angstroms)
+            radius within which contacts exist
+        method: string
+            either 'cutoff' or 'best-hummer'
+
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+
+        **kwargs : argument list passed to cut-off function
+            check respective functions for reasonable values
+
+        Attributes
+        ----------
+        results: list
+            Fraction of native contacts for each frame
+        """
+        if method == 'cutoff':
+            self.fraction_contacts = hard_cut
+        elif method == 'best-hummer':
+            self.fraction_contacts = best_hummer_q
+        else:
+            self.fraction_contacts = method
+
+        # setup boilerplate
+        self.u = u
+        self._setup_frames(self.u.trajectory, start, stop, step)
+
+        self.selection = selection
+        self.grA = u.select_atoms(selection[0])
+        self.grB = u.select_atoms(selection[1])
+
+        # contacts formed in reference
+        refA, refB = refgroup
+        self.r0 = distance_array(refA.positions, refB.positions)
+        self.initial_contacts = contact_matrix(self.r0, radius)
+
+        self.fraction_kwargs = kwargs
+        self.timeseries = []
+
+    def _single_frame(self):
+        # compute distance array for a frame
+        d = distance_array(self.grA.positions, self.grB.positions)
+
+        # select only the contacts that were formed in the reference state
+        r = d[self.initial_contacts]
```

Comment:
Could we make the distance_array call above lighter by applying the mask there instead of here? (`grA[self.initial_contacts].positions`)